tests/conftest.py

```
@@ -2125,6 +2128,102 @@ def additional_testsuite_properties(record_testsuite_property, pytestconfig):
     for dr_operator_name, dr_operator_version in dr_operator_versions.items():
         record_testsuite_property(f"rp_{dr_operator_name}", dr_operator_version)
 
+    yield
+
+    _update_testsuite_properties_after_upgrade(pytestconfig)
+
+
+def _update_testsuite_properties_after_upgrade(pytestconfig):
```

Ruff (0.15.21)

[warning] Too many branches (17 > 12)

(PLR0912)